homeassistant/components/smlight/__init__.py

```diff
@@ -0,0 +1,29 @@
+"""SMLIGHT SLZB Zigbee device integration."""
+
+from __future__ import annotations
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+
+from .coordinator import SmDataUpdateCoordinator
+
+PLATFORMS: list[Platform] = [
+    Platform.SENSOR,
+]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
```

Comment:
You are missing a custom type for your entry like `type SmConfigEntry = ConfigEntry[SmDataUpdateCoordinator]`